strider/trapi.py

```diff
@@ -390,3 +391,112 @@ def build_unique_kg_edge_ids(message: Message):
                 for eb in edge_binding_list:
                     if eb["id"] == edge_id:
                         eb["id"] = new_edge_id
+
+
+def is_valid_node_binding(message, nb, qgraph_node):
+    """
+    Check whether the given kgraph node
+    satisifies the given qgraph node
+    """
+    if qgraph_node.get("id", None) is not None:
+        if nb["id"] not in qgraph_node["id"]:
+            return False
+
+    kgraph_node = message["knowledge_graph"]["nodes"][nb["id"]]
+    if qgraph_node.get("category", None) is not None:
+        # Build list of allowable categories for kgraph nodes
+        qgraph_allowable_categories = []
+        for c in ensure_list(qgraph_node["category"]):
```

Comment:
I think we already did `standardize_graph_lists` and shouldn't need to `ensure_list` here, right?

Reply:
The `standardize_graph_lists` is called on the knowledge graph, and `ensure_list` is working on a qgraph node.